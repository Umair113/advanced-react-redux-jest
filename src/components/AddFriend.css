:local(.flex) {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin-top: 5px;
}
