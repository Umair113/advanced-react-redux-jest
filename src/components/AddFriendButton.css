:local(.addFriend) {
  margin: 0 10px;
}
