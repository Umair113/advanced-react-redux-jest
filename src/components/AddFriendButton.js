import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './AddFriendButton.css';


function AddFriendButton({ onButtonSubmit }) {
  const handleSubmit = (e) => {
    e.stopPropagation();
    onButtonSubmit();
  };

  return(
    <button
    className={classnames('btn btn-primary', styles.addFriend)}
    onClick={handleSubmit}
    >
    Add Friend
  </button>);
}

AddFriendButton.propTypes = {
  onButtonSubmit: PropTypes.func.isRequired
};

export default AddFriendButton