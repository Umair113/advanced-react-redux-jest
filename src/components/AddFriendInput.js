import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './AddFriendInput.css';

function AddFriendInput({ name, onInputChange, onInputSubmit }) {
  const handleChange = (e) => {
    onInputChange(e.target.value);
  };

  const handleSubmit = (e) => {
    const name = e.target.value.trim();
    if (e.which === 13) {
      onInputSubmit(name);
    }
  };

  return (
    <input
      type="text"
      autoFocus="true"
      className={classnames('form-control', styles.addFriendInput)}
      placeholder="Type the name of a friend"
      value={name}
      onChange={handleChange}
      onKeyDown={handleSubmit} />
  );
}

AddFriendInput.propTypes = {
  name: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
  onInputSubmit: PropTypes.func.isRequired
};

export default AddFriendInput
