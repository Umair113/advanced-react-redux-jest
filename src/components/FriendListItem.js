import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './FriendListItem.css';

function FriendListItem({ id, name, gender, starred, starFriend, deleteFriend }) {

  const handleStar = () => { starFriend(id); };
  const handleDelete = () => { deleteFriend(id); };

  return (
    <li className={styles.friendListItem}>
      <div className={styles.friendInfos}>
        <div><span>{name}</span></div>
        <div><span>{capitalize(gender)}</span></div>
        <div>
          <small>xx friends in common</small>
        </div>
      </div>
      <div className={styles.friendActions}>
        <button className={`btn btn-default ${styles.btnAction}`}
                onClick={handleStar}>
          <i className={classnames('fa', {
            'fa-star': starred,
            'fa-star-o': !starred
          })} />
        </button>
        <button className={`btn btn-default ${styles.btnAction}`}
                onClick={handleDelete}>
          <i className="fa fa-trash" />
        </button>
      </div>
    </li>
  );
}

function capitalize(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}

FriendListItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  starred: PropTypes.bool,
  starFriend: PropTypes.func.isRequired,
  deleteFriend: PropTypes.func.isRequired
};

export default FriendListItem
