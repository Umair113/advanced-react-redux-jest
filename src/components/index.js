export { default as AddFriendInput } from './AddFriendInput';
export { default as FriendList } from './FriendList';
export { default as FriendListItem } from './FriendListItem';
export { default as SelectFriendGender } from './SelectFriendGender';
export { default as AddFriend } from './AddFriend';
export { default as AddFriendButton } from './AddFriendButton';
export { default as PageFlipper } from './PageFlipper';