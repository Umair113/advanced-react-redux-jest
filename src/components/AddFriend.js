import React, { Component } from 'react';
import { AddFriendInput, SelectFriendGender, AddFriendButton } from '../components';
import PropTypes from 'prop-types';
import styles from './AddFriend.css';

class AddFriend extends Component {
  render () {
    return (
      <div>
        <AddFriendInput name={this.state.name} onInputChange={this.handleInputChange} onInputSubmit={this.handleInputSubmit} />
        <div className={styles.flex}>
          <SelectFriendGender gender={this.state.gender} onSelect={this.handleSelectChange}/>
          <AddFriendButton onButtonSubmit={this.handleButtonSubmit}/>
        </div>
      </div>
    );
  }

  constructor(props) {
    super(props);
    this.state = {
      name: '',
      gender: 'male'
    };

    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleInputSubmit = this.handleInputSubmit.bind(this);
    this.handleSelectChange = this.handleSelectChange.bind(this);
    this.handleButtonSubmit = this.handleButtonSubmit.bind(this)
  }

  handleInputChange (input) {
    this.setState({ name: input });
  }

  handleSelectChange(gender) {
    this.setState({ gender });
  }

  handleInputSubmit (name) {
    if (name.length > 0) {
      this.addFriend(name);
      this.resetFields();
    }
  }

  handleButtonSubmit() {
    if (this.state.name.length > 0) {
      this.addFriend();
      this.resetFields();
    }
  }

  resetFields() {
    this.setState({
      name: '',
      gender: 'male'
    });
  }

  addFriend(name = this.state.name, gender = this.state.gender) {
    this.props.addFriend(name, gender);
  }
}

AddFriend.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriend;