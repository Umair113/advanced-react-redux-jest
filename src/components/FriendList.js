import React from 'react';
import PropTypes from 'prop-types';
import styles from './FriendList.css';
import FriendListItem from './FriendListItem';

function FriendList({ friends, starFriend, deleteFriend, page, pageSize }) {
  let baseId = pageSize * (page - 1);
  return (
    <ul className={styles.friendList}>
      {
        friends.map((friend, index) => {
          return (
            <FriendListItem
              key={baseId + index}
              id={baseId + index}
              name={friend.name}
              gender={friend.gender}
              starred={friend.starred}
              starFriend={starFriend}
              deleteFriend={deleteFriend}
            />
          );
        })
      }
    </ul>
  );
}

FriendList.propTypes = {
  friends: PropTypes.array.isRequired,
  starFriend: PropTypes.func.isRequired,
  deleteFriend: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired
};

export default FriendList;
