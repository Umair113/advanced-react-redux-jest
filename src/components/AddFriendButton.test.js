import React from 'react';
import renderer from 'react-test-renderer';
import { AddFriendButton } from '../components';

describe('AddFriendButton Component Test', () => {
  const noop = () => {};

  it('renders as expected', () => {
    const componentTree = renderer.create(
      <AddFriendButton
        name={'John Snow'}
        onButtonSubmit={noop}
      />)
    .toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});