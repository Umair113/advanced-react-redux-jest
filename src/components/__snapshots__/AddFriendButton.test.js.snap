// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFriendButton Component Test renders as expected 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
>
  Add Friend
</button>
`;
